golem-component-service-base/src/service/component.rs

```diff
@@ -1343,6 +1367,52 @@ impl<Owner: ComponentOwner, Scope: PluginScope> ComponentService<Owner>
         }
     }
 
+    async fn get_file_contents(
+        &self,
+        component_id: &ComponentId,
+        version: ComponentVersion,
+        path: &str,
+        owner: &Owner,
+    ) -> Result<BoxStream<'static, Result<Bytes, ComponentError>>, ComponentError> {
+        let component = self
+            .get_by_version(
+                &VersionedComponentId {
+                    component_id: component_id.clone(),
+                    version,
+                },
+                owner,
+            )
+            .await?;
+        if let Some(component) = component {
+            info!(owner = %owner, component_id = %component.versioned_component_id, "Stream component file: {}", path);
+
+            let file = component
+                .files
+                .iter()
+                .find(|&file| file.path.to_rel_string() == path)
+                .ok_or(ComponentError::InvalidFilePath(path.to_string()))?;
+
+            let stream = self
+                .component_files_service
+                .get(&owner.account_id(), &file.key)
+                .await
+                .map_err(|e| {
+                    ComponentError::initial_component_file_upload_error(
+                        "Failed to get component file",
+                        e,
+                    )
+                })?
+                .ok_or(ComponentError::FailedToDownloadFile)?;
+            let mapped_stream = stream.map_err(|e| {
```

Comment:
Very minor. I would just inline this with the above rather than calling it a mapped_stream. It's just mapping error.